### PR TITLE
feat(GDB-10855) Implement Cluster Creation Functionality

### PR DESCRIPTION
**Pull request by teodossidossev**
## What:
- Implemented functionality to create a new cluster in the cluster management interface.
- Added a form to define the initial cluster configuration
- Introduced form validation to ensure that the cluster configuration is valid before saving.
- Added logic to distinguish between creating a new cluster and editing an existing cluster.

## Why:
Requirements

## How:
- Extended the existing cluster management interface by adding a new form for defining a cluster's initial configuration.
- Implemented validation logic for essential cluster parameters.
- Updated the ClusterContextService and ClusterViewModel to handle the creation of new clusters and validate the configuration.
- Created new methods in the cluster-management.controller.js for handling the cluster creation process.
- Updated the UI to provide advanced configuration options.

### CODE REVIEW

File: src/js/angular/clustermanagement/directives/cluster-list.directive.js [hunk 1]
```diff
@@ -1,4 +1,16 @@
-import {Location} from "../../models/clustermanagement/cluster";
+import {ClusterConfiguration, Location} from "../../models/clustermanagement/cluster";
+
+const getAdvancedOptionsClass = function () {
```
Comment (boyan-tonchev) [inline src/js/angular/clustermanagement/directives/cluster-list.directive.js:3]: Why is this function declared here, and later a variable is created with a reference to it? Why not simply create a public function `getAdvancedOptionsClass` within the directive? By making it a public function, you can use the second argument of the link method, `element`, to check if there is an `optionsModule`.

File: src/js/angular/clustermanagement/controllers/cluster-management.controller.js [hunk 2]
```diff
@@ -177,24 +162,7 @@ function ClusterManagementCtrl($scope, $http, $q, toastr, $repositories, $uibMod
     };
 
     $scope.showCreateClusterDialog = function () {
```
Comment (boyan-tonchev) [inline src/js/angular/clustermanagement/controllers/cluster-management.controller.js:164]: You can remove this function, and where it is used, simply call $scope.showUpdateClusterGroupDialog() directly.

Comment (teodossidossev) [reply src/js/angular/clustermanagement/controllers/cluster-management.controller.js]: done

File: src/js/angular/clustermanagement/directives/cluster-list.directive.js (hunk 1 above)
Comment (teodossidossev) [reply src/js/angular/clustermanagement/directives/cluster-list.directive.js]: done

File: src/js/angular/clustermanagement/directives/cluster-list.directive.js [hunk 3]
```diff
@@ -1,4 +1,16 @@
-import {Location} from "../../models/clustermanagement/cluster";
+import {ClusterConfiguration, Location} from "../../models/clustermanagement/cluster";
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/js/angular/clustermanagement/directives/cluster-list.directive.js]: removed

File: src/js/angular/models/clustermanagement/cluster.js [hunk 4]
```diff
@@ -399,13 +419,37 @@ export class ClusterViewModel {
...
         const addNodes = Array.from(this._addToCluster.values()).map((node) => node.rpcAddress || node.address);
         const removeNodes = Array.from(this._deleteFromCluster.values()).map((node) => node.rpcAddress || node.address);
-        return {
+        const updateActions = {
             addNodes,
             removeNodes
         };
+
+        if (!this.hasCluster()) {
+            this._clusterConfiguration.nodes = addNodes;
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/js/angular/models/clustermanagement/cluster.js:453]: extended to a new method

File: src/js/angular/models/clustermanagement/cluster.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import {value} from "lodash/seq";
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/js/angular/models/clustermanagement/cluster.js]: deleted

**Review by boyan-tonchev — APPROVED** [review]

**Pull request merged**